demo/dask/dask_custom_eval_demo.py

```diff
@@ -0,0 +1,45 @@
+from dask.distributed import Client, LocalCluster
+import dask.dataframe as dd
+import dask.array as da
+import numpy as np
+import xgboost as xgb
+
+
+# Define rmse function using Python that mimics the internal xgboost rmse function
+def custom_rmse(preds, dtrain):
+    labels = dtrain.get_label()
+    weights = dtrain.get_weight()
+
+    mse_local_sum = sum(np.square(labels - preds) * weights)
+    weights_local_sum = sum(weights)
+    # Pass the sum mse and weights to an allreduce call as a numpy array
+    result = xgb.rabit.allreduce(np.asarray([mse_local_sum, weights_local_sum]), 2)

```

Comment:
@CodingCat My question is why calling these 2 lines requires that much of code in #4280 .  Personally I prefer the code in this PR @RAMitchell  since it seems much simpler,  I glanced through the code and understood how to do it immediately.  And I believe I can debug my code much easier since I can just put a print in here and get to know everything.

Revisiting comments in #4280 I get that you want to mask the complexity in dealing with rabit and multiclass/rank metrics @hcho3 .  I like the idea of making things simple for users, but can the layer be done as simple as these two lines?

Reply:
@CodingCat Also I think there's no need to mask the `sum` operation in #4280 .  What if I want to use `mean` or `max` instead of `sum`?

Reply:
@trivialfis I think the user can define `getFinal()` to implement a custom reduce op

Reply:
Actually no, `getFinal()` is not a custom reduce op. Sorry about that. My brain is fried after long day of work